### ISSUE TITLE
improve ica.plot_overlay for raw instances

### ISSUE THREAD

**Issue by vferat** [mne-tools/mne-python#11061]
#### Describe the new feature or enhancement
Currently when providing `inst=raw`, [`ica.plot_overlay`](https://mne.tools/stable/generated/mne.preprocessing.ICA.html?highlight=plot+overlay#mne.preprocessing.ICA.plot_overlay) transform it to an `evoked` for display. This doesn't really make sense when working with spontaneous data.
I would be great to be able to plot the overlay on the classic `raw.plot()` window. Each data channel would be represented by two superimposed traces of different colors: one representing the raw instance and the other the raw instance without the ICA components.
It would be useful as in [this example](https://mne.tools/stable/auto_tutorials/preprocessing/40_artifact_correction_ica.html#selecting-ica-components-manually), where we users are forced to manually apply ICA to a copy of the raw instance and then plot the 2 raw instances.

#### Describe your proposed implementation
Directly change the behavior of` ica.plot_overlay(insta=raw)` or create something like joint_plot(raw1, raw2) which allows to plot several raw instances with same info on the same plot.

#### Describe possible alternatives
A way to synchronize two raw.plot() windows, which allows to have always the same time window of the recordings in view.

#### Additional comments
This feature could also be implemented in` mne-qt-browser`

**Comment by larsoner** [comment]
I agree this would be useful. From an implementation standpoint it seems difficult, though, since our raw plotting is based on working with a single instance.
